core/server/middleware/error-handler.js

```diff
@@ -51,9 +51,16 @@ _private.prepareError = function prepareError(err, req, res) {
     }
 
     if (!(err instanceof errors.GhostError)) {
-        err = new errors.GhostError({
-            err: err
-        });
+        // We need a special case for 404 errors
+        if (err.statusCode && err.statusCode === 404) {
```

Comment:
yeah we do not inherit the status code from the error to inherit from, which makes sense in general.
https://github.com/TryGhost/Ghost/blob/master/core/server/errors.js#L44

but in case of NOT an instance of GhostError we can override the statusCode in all cases:
`new errors.GhostError({err: err, statusCode: err.statusCode})`